restli-common/src/main/java/com/linkedin/restli/common/validation/RestLiDataSchemaDataValidator.java

```diff
@@ -53,6 +57,8 @@ public RestLiDataSchemaDataValidator(Annotation[] annotations,
     }
 
     _validatingSchema = validatingSchema;
+    _inputDataValidator = new DataValidator(_validatingSchema);
```

Comment:
This is not going to be useful. The validatingSchema is for output, based on projection.
and for inputs it would be the resource entity schema or a subset based on patch for partial_update.

Instead of initializing in the constructor, you should just use getValidatorFor* methods, but cache them after first invocation? 

Also this validator class is used only for output validation (input uses RestLiDataValidator directly). so you can remove all the input validator logic and remove the override for getValidatorForInput

Reply:
I actually don't see the need for changes in this file. The changes in other two classes should be enough to override the validator construction logic.

Reply:
Is the suggestion to sub-class `RestLiDataSchemaDataValidator` and `RestLiDataValidator` ourselves in our own implementation and use them in our sub-class of `RestLiValidator` by overriding `createRequestRestLiDataValidator` and `createResponseRestLiDataValidator`?

For the output use case, is there a reason we need to do it explicitly via a cache? For each `_validatingSchema` instantiation, it should be the same `DataSchemaAnnotationValidator` from my understanding, so it feels that this doesn't have to be done via a cache on an outer layer

Reply:
If you plan to re-use this validator, then we can leave the caching. However, this validator is used only for output validation (maybe the javadoc should be updated to mention this). So you should remove the override for getValidatorForInput.

Reply:
Removed the override and updated the javadoc